pandas/_libs/src/vendored/ujson/python/objToJSON.c

```diff
@@ -1280,13 +1280,10 @@ char **NpyArr_encodeLabels(PyArrayObject *labels, PyObjectEncoder *enc,
     NPY_DATETIMEUNIT dateUnit = NPY_FR_ns;
     if (PyTypeNum_ISDATETIME(type_num)) {
       is_datetimelike = 1;
-      PyArray_VectorUnaryFunc *castfunc =
-          PyArray_GetCastFunc(PyArray_DescrFromType(type_num), NPY_INT64);
-      if (!castfunc) {
-        PyErr_Format(PyExc_ValueError, "Cannot cast numpy dtype %d to long",
-                     enc->npyType);
-      }
-      castfunc(dataptr, &i8date, 1, NULL, NULL);
+      PyObject *scalarItem = PyArray_ToScalar(dataptr, labels);
+      PyArray_CastScalarToCtype(scalarItem, &i8date,
+                                PyArray_DescrFromType(NPY_INT64));
+      Py_DECREF(scalarItem);
```

Comment:
Here, you can just use:
```
memcpy(&i8date, dataptr, sizeof(i8data));
```
I think, since you know this is an int64, as datetimes are int64 (I am unsure about alignment, thus the memcpy).

Same below?

Reply:
Wow, this is even easier, thanks!

Reply:
Actually, I just realized the cast function doesn't handle alignment, so even `i8data = *(npy_int64 *)dataptr;` should work.